docs/xcompilationthreads.md

```diff
@@ -31,7 +31,16 @@ Use this option to specify the number of compilation threads that are used by th
 
         -XcompilationThreads<n>
 
-: where `<n>` is the number of threads, in the range 1-7 inclusive. Any number outside this range is ignored. 
+: where `<n>` is the number of threads, in the range 1-7 inclusive. Any number outside this range is ignored.
```

Comment:
We should probably not mention about JVM client here, since if the JVM is not attaching to any client the behavior is the same: for Java8 and above, the maximum number of usable compilation threads is 15 (not 16!).  The default number of threads is automatically determined by the JVM. We can talk about JITServer here, but I don't think we need to explain why somebody might want to use a larger number of threads.

Reply:
@AlexeyKhrabrov the topic was updated in 2019 to say that the maximum was changed to 7 (see https://github.com/eclipse-openj9/openj9-docs/issues/435). Seems pretty clear from that issue that it was a maximum value? And now in 0.33 it's changed again to 15 - as I understand it, this change applies on a JITServer client and when using the JIT compiler "normally", ie without enabling the JITServer feature. Though things are confusing as the issue says that the change applies only to Linux.

Actually, looking at the code, looks like back in 2019 there was just MAX_CLIENT_USABLE_COMP_THREADS (7/63 for client/server) whereas now there's DEFAULT_CLIENT_USABLE_COMP_THREADS with the values from the old MAX_CLIENT_USABLE_COMP_THREADS (ie 7/63), and new values for MAX_CLIENT_USABLE_COMP_THREADS (15/999).

I've asked Sreekala to add words to clarify that the text later on about a max value of 999 applies to a JITServer server only (although I can't see my comments in github for some reason!) but perhaps we should add a table given that there are now 2 default values, depending on scenario.

Reply:
Let me try to clarify the current state of affairs for Java8 and above (IBM SDK with Java7 does not apply).

The default number of compilation threads is determined by JIT heuristics. For JITServer the default is 63.
The maximum number of compilation threads is 15. For JITServer the maximum value is 999.

JITServer is only available on Linux platforms (X/P/Z).

Reply:
Thanks @mpirvu. And what are the default & max values for JIT running without the JITServer feature enabled? Same as for JITServer client presumably, even on non-Linux platforms? eg if you're running JIT on AIX or on Linux in non-JITServer mode.

Reply:
Indeed, it does not matter if we are running in client mode or vanilla OpenJ9, the values are the same.